File: 250220/아름다운 수/beautiful-number.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public static int n;
    static int count = 0;
    static ArrayList<Integer> list = new ArrayList<>();

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();

        // Please write your code here.
        choose(1);
        System.out.println(count);

    }
    
    static void choose(int position){
        if(n < position){
            if(n+1 == position) {
                count++;
            }
            return;
        }

        for(int  i = 1;  i <= 4; i++){
            for(int j = 0; j < i; j++){
                list.add(i);
            }
            choose(list.size() +1);
            for(int j = list.get(list.size() -1); 0 < j; j--){
                list.remove(list.size() -1);
            }
        }

    }
}

File: 260325/아름다운 수/beautiful-number.py
N = int(input())

# Please write your code here.
nums = []
answer = 0
def choose(n):
    global answer
    if n == N:
        answer += 1
        return
    elif n > N:
        return
    
    for k in range(1, 4+1):
        for kk in range(k):
            nums.append(k)
        choose(n + k)
        for kk in range(k):
            nums.pop()


choose(0)
print(answer)

